tools/fetch-lua-type-definitions.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+base_dir="$( cd "$(dirname "$0")/.." >/dev/null 2>&1 ; pwd -P )"
```

Comment:
You do permanently change the directory here, don't you?
Consider using `pushd` and later `popd`